actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -248,11 +248,18 @@ def render_partial_template(view, locals, template, layout, block)
           identifier: template.identifier,
           layout: layout && layout.virtual_path
         ) do |payload|
-          content = template.render(view, locals, add_to_stack: !block) do |*name|
-            view._layout_for(*name, &block)
+          buffer = view.output_buffer
+          content = buffer.capture do
```

Comment:
Note to self: Ideally when rendering a partial we'd receive the "parent" template buffer and directly write into it, so no need to capture and concatenate later.

Reply:
Actually scratch that, it would change the `render` semantic too much.